### ISSUE TITLE
evaluation on NYU looks horrible

### ISSUE THREAD

**Issue by st2yang** [DrSleep/light-weight-refinenet#34]
Hi Vladimir,

Thanks for your great work and generous release. Your tutorial to reproduce the results is very detailed, and I appreciate it a lot.

I am trying to use your code in my research project. First I tried to re-train the network on NYUD with the provided code, the IoU is 0.419 which is fine. Then I tried to visualize the prediction as you did in the notebook. But the result looks pretty bad, and I don't know why the predicted mask image is green.

I tried evaluation with given weight file and the file I trained myself, the difference is huge as follows. The only difference I made in the code between two result figures is loading different weight file.

Do you have an idea why this could happen?

Thank you in advance!

<img width="983" alt="Screen Shot 2019-04-18 at 10 58 36 PM" src="https://user-images.githubusercontent.com/32805298/56404661-866f4400-622d-11e9-834c-d5ddd37b437b.png">

**Comment by DrSleep** [comment]
that looks weird.

just to clarify: did you run the exactly same notebook code? and only changed the checkpoint file?
if so, I wonder if you can share the checkpoint file and the training log if possible.

**Comment by st2yang** [comment]
Hi Vladimir,

Thanks for your reply.

Yes I run the code adapted from the notebook (mainly just copied and pasted). I can produce the right result figure with provided weights file. And I only changed the weights file then the result figure becomes horrible.

I packed related files (infer.py adapted from example notebook, training log and weights file) here https://drive.google.com/open?id=1SOf1CFudGdZ0Tf7OHASoBQu3ta22BJo-
can you please take a look?

Thanks a lot!

**Comment by DrSleep** [comment]
I did not run the models yet.

But I think I spotted the error: if you train using the script in this repo, then the model is being wrapped with nn.DataParallel - https://github.com/DrSleep/light-weight-refinenet/blob/master/src/train.py#L354
Now, when it happens, parameters' names in the state_dict have 'module.' before their original names. 
In your inference script, you don't wrap the model with nn.DataParallel, hence the pre-trained weights are not actually loaded. You can see that by setting up `strict=True`

TL;DR in your inference script apply nn.DataParallel to your model before loading the weights.

**Comment by st2yang** [comment]
You're right. I made some mistakes in loading the weight file.

After fixing it, now I can get reasonably good prediction result (for ResNet50) as follows.

Thanks a lot for your help. I appreciate it a lot!

![Screenshot from 2019-04-19 09-37-21](https://user-images.githubusercontent.com/32805298/56428940-c7477700-6286-11e9-8c30-dd92e96925ac.png)

**Comment by xiaoyuamw** [comment]
Hi@st2yang

Can you tell me what you had changed to solve you issue? and the link "https://drive.google.com/open?id=1SOf1CFudGdZ0Tf7OHASoBQu3ta22BJo-" provided by you can't to be opened now, so i also don't know what you have changed!
Thank you very much.

Best Regards.

**Comment by st2yang** [comment]
@xiaoyuamw You can refer to Vladimir's response above or this repo https://github.com/choicelab/grasping-invisible.

As far I can remember, I just warp the model with nn. DataParallel
